roles/ruby_s/tasks/main.yml

```diff
@@ -0,0 +1,65 @@
+---
+- name: ruby | check version
+  ansible.builtin.shell: ruby -v | awk '{print $2}'
+  register: installed_ruby
+  ignore_errors: true
+
+- name: prints out the installed_ruby variables
+  ansible.builtin.debug:
+    msg: ruby version "{{ installed_ruby.stdout }}"
+
+- include_tasks: cleanup.yml
+  when: installed_ruby.stdout != "{{ desired_ruby_version }}"
+
+- name: ruby |lookup download path and checksum from ruby-lang release index
+  ansible.builtin.shell: curl https://cache.ruby-lang.org/pub/ruby/index.txt | grep {{ ruby_version }}.tar.gz
+  register: ruby_index_line
+
+- name: ruby | separate index entry values
+  ansible.builtin.set_fact:
+    ruby_values: "{{ ruby_index_line.stdout | regex_findall('(\\S+)') }}"
+
+- name: get ruby url
+  ansible.builtin.set_fact:
+    ruby_gzip_url: "{{ ruby_values[1] }}"
+
+- name: get ruby checksum
+  set_fact:
+    ruby_sha_256: "{{ ruby_values[3] }}"
+
+- name: ruby | setup install directory
+  ansible.builtin.set_fact:
+    install_path: "~/install"
```

Comment:
Why use `set_fact` for the install path? That's a value on the local machine. We can set a role default and let each project override it if they want. I might choose a default value that's not dependent on the user (something like `/opt/install`, maybe?).